cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/DRepDeposits.hs

```diff
@@ -0,0 +1,254 @@
+                               , "--key-reg-deposit-amt", show @Integer depositAmount
+                               , "--out-file", registrationCertificateFile dRepRegistrationCertificate
+                               ]
+  return dRepRegistrationCertificate
+
+-- DRep registration transaction composition (without signing)
+
+newtype TxBody = TxBody { txBodyFile :: FilePath }
+
+createDRepRegistrationTxBody
+  :: (H.MonadAssertion m, MonadTest m, MonadCatch m, MonadIO m)
+  => H.ExecConfig
+  -> EpochStateView
+  -> ShelleyBasedEra era
+  -> FilePath
+  -> String
+  -> DRepRegistrationCertificate
+  -> PaymentKeyInfo
+  -> m TxBody
+createDRepRegistrationTxBody execConfig epochStateView sbe work prefix drepRegCert wallet = do
+  let dRepRegistrationTxBody = TxBody (work </> prefix <> ".txbody")
+  walletLargestUTXO <- findLargestUtxoForPaymentKey epochStateView sbe wallet
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "build"
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr wallet
+    , "--tx-in", Text.unpack $ renderTxIn walletLargestUTXO
+    , "--certificate-file", registrationCertificateFile drepRegCert
+    , "--witness-override", show @Int 2
+    , "--out-file", txBodyFile dRepRegistrationTxBody
+    ]
+  return dRepRegistrationTxBody
+
+-- Transaction signing
+
+newtype SignedTx = SignedTx { signedTxFile :: FilePath }
+
+signTx :: (MonadTest m, MonadCatch m, MonadIO m)
+  => H.ExecConfig
+  -> FilePath
+  -> String
+  -> TxBody
+  -> [PaymentKeyPair]
+  -> m SignedTx
+signTx execConfig work prefix txBody signatoryKeyPairs = do
+  let signedTx = SignedTx (work </> prefix <> ".tx")
+  void $ H.execCli' execConfig $
+    [ "conway", "transaction", "sign"
+    , "--tx-body-file", txBodyFile txBody
+    ] ++ (concat [["--signing-key-file", paymentSKey kp] | kp <- signatoryKeyPairs]) ++
+    [ "--out-file", signedTxFile signedTx
+    ]
+  return signedTx
+
+-- Transaction submission
+
+submitTx
+  :: (MonadTest m, MonadCatch m, MonadIO m)
+  => H.ExecConfig
+  -> SignedTx
+  -> m ()
+submitTx execConfig signedTx =
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "submit"
+    , "--tx-file", signedTxFile signedTx
+    ]
+
+-- Attempt to submit transaction that must fail
+
+failToSubmitTx
+  :: (MonadTest m, MonadCatch m, MonadIO m)
+  => H.ExecConfig
+  -> SignedTx
+  -> m ()
+failToSubmitTx execConfig signedTx = GHC.withFrozenCallStack $ do
+  (exitCode, _, _) <- H.execFlexAny' execConfig "cardano-cli" "CARDANO_CLI"
+                                     [ "conway", "transaction", "submit"
+                                     , "--tx-file", signedTxFile signedTx
+                                     ]
+  case exitCode of
+    ExitSuccess -> H.failMessage GHC.callStack "Transaction submission was expected to fail but it succeeded"
+    _ -> return ()
+
+-- Obtains the amounts of the DRep deposits given some assumptions
+
```

Comment:
Can you make that an haddock?

Also implies to other functions above

Reply:
Addressed [here](https://github.com/IntersectMBO/cardano-node/pull/5748/commits/31c5e98fe7aa8f3ded0d0042ebb1e0d5d057725f)